### ISSUE TITLE
Static build uhd-3.13.1.0-rc1 failed with boost_1_66_0 in the same place as six months ago with previous releases

### ISSUE THREAD

**Issue by amore-amore** [EttusResearch/uhd#223]
```console 
...
CMake Warning at /usr/local/share/cmake/Modules/FindBoost.cmake:801 (message):
  New Boost version may have incorrect or missing dependencies and imported
  targets
Call Stack (most recent call first):
  /usr/local/share/cmake/Modules/FindBoost.cmake:907 (_Boost_COMPONENT_DEPENDENCIES)
  /usr/local/share/cmake/Modules/FindBoost.cmake:1542 (_Boost_MISSING_DEPENDENCIES)
  CMakeLists.txt:343 (FIND_PACKAGE)

CMake Warning at /usr/local/share/cmake/Modules/FindBoost.cmake:801 (message):
  New Boost version may have incorrect or missing dependencies and imported
  targets
Call Stack (most recent call first):
  /usr/local/share/cmake/Modules/FindBoost.cmake:907 (_Boost_COMPONENT_DEPENDENCIES)
  /usr/local/share/cmake/Modules/FindBoost.cmake:1542 (_Boost_MISSING_DEPENDENCIES)
  CMakeLists.txt:343 (FIND_PACKAGE)

-- Boost version: 1.66.0
-- Found the following Boost libraries:
--   chrono
--   date_time
--   filesystem
--   program_options
--   regex
--   system
--   unit_test_framework
--   serialization
--   thread
-- Boost include directories: /home/build/uhd-3.13.1.0-rc1/host/build/boost/include
-- Boost library directories: /home/build/uhd-3.13.1.0-rc1/host/build/boost/lib
-- Boost libraries: /home/build/uhd-3.13.1.0-rc1/host/build/boost/lib/libboost_chrono-mt.a;/home/build/uhd-3.13.1.0-rc1/host/build/boost/lib/libboost_date_time-mt.a;/home/build/uhd-3.13.1.0-rc1/host/build/boost/lib/libboost_filesystem-mt.a;/home/build/uhd-3.13.1.0-rc1/host/build/boost/lib/libboost_program_options-mt.a;/home/build/uhd-3.13.1.0-rc1/host/build/boost/lib/libboost_regex-mt.a;/home/build/uhd-3.13.1.0-rc1/host/build/boost/lib/libboost_system-mt.a;/home/build/uhd-3.13.1.0-rc1/host/build/boost/lib/libboost_unit_test_framework-mt.a;/home/build/uhd-3.13.1.0-rc1/host/build/boost/lib/libboost_serialization-mt.a;/home/build/uhd-3.13.1.0-rc1/host/build/boost/lib/libboost_thread-mt.a
--
-- Python checking for Python version 2.7 or greater
-- Python checking for Python version 2.7 or greater - found
--
-- Python checking for Mako templates 0.4.2 or greater
-- Python checking for Mako templates 0.4.2 or greater - found
--
-- Python checking for requests 2.0 or greater
-- Python checking for requests 2.0 or greater - "import requests" failed
--
-- Python checking for numpy 1.7 or greater
-- Python checking for numpy 1.7 or greater - found
--
-- Configuring LibUHD support...
--   Dependency Boost_FOUND = 1
--   Dependency HAVE_PYTHON_PLAT_MIN_VERSION = TRUE
--   Dependency HAVE_PYTHON_MODULE_MAKO = TRUE
--   Enabling LibUHD support.
--   Override with -DENABLE_LIBUHD=ON/OFF
--
-- Configuring LibUHD - C API support...
--   Dependency ENABLE_LIBUHD = ON
--   Enabling LibUHD - C API support.
--   Override with -DENABLE_C_API=ON/OFF
--
-- Configuring LibUHD - Python API support...
--   Dependency ENABLE_LIBUHD = ON
--   Dependency BOOST_PYTHON_FOUND = ON
--   Dependency HAVE_PYTHON_MODULE_NUMPY = TRUE
--   Dependency PythonLibs_FOUND = TRUE
--   Enabling LibUHD - Python API support.
--   Override with -DENABLE_PYTHON_API=ON/OFF
--
-- Configuring Examples support...
--   Dependency ENABLE_LIBUHD = ON
--   Enabling Examples support.
--   Override with -DENABLE_EXAMPLES=ON/OFF
--
-- Configuring Utils support...
--   Dependency ENABLE_LIBUHD = ON
--   Enabling Utils support.
--   Override with -DENABLE_UTILS=ON/OFF
--
-- Configuring Tests support...
--   Dependency ENABLE_LIBUHD = ON
--   Enabling Tests support.
--   Override with -DENABLE_TESTS=ON/OFF
--
-- Checking for module 'libgps'
--
-- Checking for module 'libgps >= 3.11'
--
-- Could NOT find LIBGPS (missing: LIBGPS_LIBRARY LIBGPS_INCLUDE_DIR)
-- Could NOT find LIBERIO (missing: LIBERIO_LIBRARY LIBERIO_INCLUDE_DIR)
--
-- Configuring LIBERIO support...
--   Dependency ENABLE_LIBUHD = ON
--   Dependency LIBERIO_FOUND = FALSE
--   Disabling LIBERIO support.
--   Override with -DENABLE_LIBERIO=ON/OFF
--
-- Configuring USB support...
--   Dependency ENABLE_LIBUHD = ON
--   Dependency LIBUSB_FOUND = TRUE
--   Enabling USB support.
--   Override with -DENABLE_USB=ON/OFF
--
-- Configuring GPSD support...
--   Dependency ENABLE_LIBUHD = ON
--   Dependency ENABLE_GPSD = OFF
--   Dependency LIBGPS_FOUND = FALSE
--   Disabling GPSD support.
--   Override with -DENABLE_GPSD=ON/OFF
--
-- Configuring B100 support...
--   Dependency ENABLE_LIBUHD = ON
--   Dependency ENABLE_USB = ON
--   Disabling B100 support.
--   Override with -DENABLE_B100=ON/OFF
--
-- Configuring B200 support...
--   Dependency ENABLE_LIBUHD = ON
--   Dependency ENABLE_USB = ON
--   Enabling B200 support.
--   Override with -DENABLE_B200=ON/OFF
--
-- Configuring E300 support...
--   Dependency ENABLE_LIBUHD = ON
--   Disabling E300 support.
--   Override with -DENABLE_E300=ON/OFF
--
-- Configuring USRP1 support...
--   Dependency ENABLE_LIBUHD = ON
--   Dependency ENABLE_USB = ON
--   Disabling USRP1 support.
--   Override with -DENABLE_USRP1=ON/OFF
--
-- Configuring USRP2 support...
--   Dependency ENABLE_LIBUHD = ON
--   Disabling USRP2 support.
--   Override with -DENABLE_USRP2=ON/OFF
--
-- Configuring X300 support...
--   Dependency ENABLE_LIBUHD = ON
--   Disabling X300 support.
--   Override with -DENABLE_X300=ON/OFF
--
-- Configuring N230 support...
--   Dependency ENABLE_LIBUHD = ON
--   Disabling N230 support.
--   Override with -DENABLE_N230=ON/OFF
--
-- Configuring MPMD support...
--   Dependency ENABLE_LIBUHD = ON
--   Enabling MPMD support.
--   Override with -DENABLE_MPMD=ON/OFF
--
-- Configuring N300 support...
--   Dependency ENABLE_LIBUHD = ON
--   Dependency ENABLE_MPMD = ON
--   Disabling N300 support.
--   Override with -DENABLE_N300=ON/OFF
--
-- Configuring E320 support...
--   Dependency ENABLE_LIBUHD = ON
--   Dependency ENABLE_MPMD = ON
--   Disabling E320 support.
--   Override with -DENABLE_E320=ON/OFF
--
-- Configuring OctoClock support...
--   Dependency ENABLE_LIBUHD = ON
--   Disabling OctoClock support.
--   Override with -DENABLE_OCTOCLOCK=ON/OFF
--
--
-- Could NOT find UDev (missing: UDEV_INCLUDE_DIR UDEV_LIBS)
--
-- Configuring priority scheduling...
--   Priority scheduling supported through pthread_setschedparam.
--   Setting thread names is not supported.
--
-- Configuring high resolution timing...
--   High resolution timing supported though microsec_clock.
--
-- Configuring module loading...
--   Module loading supported through dlopen.
--
-- USB support enabled via libusb.
--
-- Configuring interface address discovery...
--   Interface address discovery supported through getifaddrs.
--
-- Loading build info.
--
-- Adding B2XX device test target
-- Adding N3XX device test target
-- Adding E32x device test target
--
-- Could NOT find UDev (missing: UDEV_INCLUDE_DIR UDEV_LIBS)
CMake Warning at utils/CMakeLists.txt:123 (MESSAGE):
  Python module `requests' not found -- uhd_images_downloader.py will not
  work without it.

CMake Warning at utils/CMakeLists.txt:124 (MESSAGE):
  You may be able to install this by running 'pip install requests'

--
-- Configuring Man Pages support...
--   Dependency GZIP_FOUND = TRUE
--   Dependency NOT_WIN32 = TRUE
--   Enabling Man Pages support.
--   Override with -DENABLE_MAN_PAGES=ON/OFF
--
-- Python checking for virtualenv
-- Python checking for virtualenv - "assert hasattr(sys, 'real_prefix')" failed
-- Utilizing the python install directory: /usr/local/lib/python2.7/site-packages
--
-- ######################################################
-- # UHD enabled components
-- ######################################################
--   * LibUHD
--   * LibUHD - C API
--   * LibUHD - Python API
--   * Examples
--   * Utils
--   * Tests
--   * USB
--   * B200
--   * MPMD
--   * Man Pages
--
-- ######################################################
-- # UHD disabled components
-- ######################################################
--   * LIBERIO
--   * GPSD
--   * B100
--   * E300
--   * USRP1
--   * USRP2
--   * X300
--   * N230
--   * N300
--   * E320
--   * OctoClock
--   * Manual
--   * API/Doxygen
--
-- Building version: 3.13.1.0-0-unknown
-- Using install prefix: /usr/local
-- Configuring done
-- Generating done
-- Build files have been written to: /home/build/uhd-3.13.1.0-rc1/host/build
[  1%] Built target uhd_rpclib
Scanning dependencies of target uhd_static

...

[ 25%] Building CXX object lib/CMakeFiles/uhd_static.dir/usrp/common/validate_subdev_spec.cpp.o
[ 25%] Building CXX object lib/CMakeFiles/uhd_static.dir/usrp/common/recv_packet_demuxer.cpp.o
[ 25%] Building CXX object lib/CMakeFiles/uhd_static.dir/usrp/common/fifo_ctrl_excelsior.cpp.o
[ 26%] Building CXX object lib/CMakeFiles/uhd_static.dir/usrp/device3/device3_impl.cpp.o
[ 26%] Building CXX object lib/CMakeFiles/uhd_static.dir/usrp/device3/device3_io_impl.cpp.o
In file included from /home/build/uhd-3.13.1.0-rc1/host/lib/usrp/device3/device3_io_impl.cpp:10:
In file included from /home/build/uhd-3.13.1.0-rc1/host/lib/usrp/device3/device3_impl.hpp:14:
In file included from /home/build/uhd-3.13.1.0-rc1/host/include/uhd/transport/bounded_buffer.hpp:11:
In file included from /home/build/uhd-3.13.1.0-rc1/host/include/uhd/transport/bounded_buffer.ipp:14:
In file included from /home/build/uhd-3.13.1.0-rc1/host/build/boost/include/boost/function.hpp:70:
In file included from /home/build/uhd-3.13.1.0-rc1/host/build/boost/include/boost/preprocessor/iteration/detail/iter/forward1.hpp:52:
In file included from /home/build/uhd-3.13.1.0-rc1/host/build/boost/include/boost/function/detail/function_iterate.hpp:14:
In file included from /home/build/uhd-3.13.1.0-rc1/host/build/boost/include/boost/function/detail/maybe_include.hpp:22:
/home/build/uhd-3.13.1.0-rc1/host/build/boost/include/boost/function/function_template.hpp:138:18: error: called object type 'nullptr_t' is not a function or function pointer
          return (*f)(BOOST_FUNCTION_ARGS);
                 ^~~~
/home/build/uhd-3.13.1.0-rc1/host/build/boost/include/boost/function/function_template.hpp:925:53: note: in instantiation of member function 'boost::detail::function::function_obj_invoker1<nullptr_t, bool,
      boost::intrusive_ptr<uhd::transport::managed_buffer> >::invoke' requested here
        { { &manager_type::manage }, &invoker_type::invoke };
                                                    ^
/home/build/uhd-3.13.1.0-rc1/host/build/boost/include/boost/function/function_template.hpp:716:13: note: in instantiation of function template specialization 'boost::function1<bool,
      boost::intrusive_ptr<uhd::transport::managed_buffer> >::assign_to<nullptr_t>' requested here
      this->assign_to(f);
            ^
/home/build/uhd-3.13.1.0-rc1/host/build/boost/include/boost/function/function_template.hpp:1061:5: note: in instantiation of function template specialization 'boost::function1<bool,
      boost::intrusive_ptr<uhd::transport::managed_buffer> >::function1<nullptr_t>' requested here
    base_type(f)
    ^
/home/build/uhd-3.13.1.0-rc1/host/lib/usrp/device3/device3_io_impl.cpp:894:13: note: in instantiation of function template specialization 'boost::function<bool
      (boost::intrusive_ptr<uhd::transport::managed_buffer>)>::function<nullptr_t>' requested here
            NULL);
            ^
/usr/include/sys/_null.h:12:14: note: expanded from macro 'NULL'
#define NULL    nullptr
                ^
1 error generated.
*** Error 1 in . (lib/CMakeFiles/uhd_static.dir/build.make:3675 'lib/CMakeFiles/uhd_static.dir/usrp/device3/device3_io_impl.cpp.o': cd /home...)
*** Error 1 in . (CMakeFiles/Makefile2:122 'lib/CMakeFiles/uhd_static.dir/all')
*** Error 1 in /home/build/uhd-3.13.1.0-rc1/host/build (Makefile:163 'all')
Running tests...
Test project /home/build/uhd-3.13.1.0-rc1/host/build
      Start  1: addr_test
 1/41 Test  #1: addr_test ........................***Failed    0.03 sec
      Start  2: buffer_test
 2/41 Test  #2: buffer_test ......................***Failed    0.02 sec
      Start  3: byteswap_test
 3/41 Test  #3: byteswap_test ....................***Failed    0.01 sec
      Start  4: cast_test
 4/41 Test  #4: cast_test ........................***Failed    0.01 sec
      Start  5: cal_container_test
 5/41 Test  #5: cal_container_test ...............***Failed    0.01 sec
      Start  6: chdr_test
 6/41 Test  #6: chdr_test ........................***Failed    0.01 sec
      Start  7: constrained_device_args_test
 7/41 Test  #7: constrained_device_args_test .....***Failed    0.01 sec
      Start  8: convert_test
 8/41 Test  #8: convert_test .....................***Failed    0.05 sec
      Start  9: dict_test
 9/41 Test  #9: dict_test ........................***Failed    0.01 sec
      Start 10: eeprom_utils_test
10/41 Test #10: eeprom_utils_test ................***Failed    0.01 sec
      Start 11: error_test
11/41 Test #11: error_test .......................***Failed    0.01 sec
      Start 12: fp_compare_delta_test
12/41 Test #12: fp_compare_delta_test ............***Failed    0.01 sec
      Start 13: fp_compare_epsilon_test
13/41 Test #13: fp_compare_epsilon_test ..........***Failed    0.01 sec
      Start 14: gain_group_test
14/41 Test #14: gain_group_test ..................***Failed    0.01 sec
      Start 15: log_test
15/41 Test #15: log_test .........................***Failed    0.01 sec
      Start 16: math_test
16/41 Test #16: math_test ........................***Failed    0.01 sec
      Start 17: narrow_cast_test
17/41 Test #17: narrow_cast_test .................***Failed    0.01 sec
      Start 18: property_test
18/41 Test #18: property_test ....................***Failed    0.01 sec
      Start 19: ranges_test
19/41 Test #19: ranges_test ......................***Failed    0.01 sec
      Start 20: sid_t_test
20/41 Test #20: sid_t_test .......................***Failed    0.01 sec
      Start 21: sensors_test
21/41 Test #21: sensors_test .....................***Failed    0.01 sec
      Start 22: soft_reg_test
22/41 Test #22: soft_reg_test ....................***Failed    0.01 sec
      Start 23: sph_recv_test
23/41 Test #23: sph_recv_test ....................***Failed    0.01 sec
      Start 24: sph_send_test
24/41 Test #24: sph_send_test ....................***Failed    0.01 sec
      Start 25: subdev_spec_test
25/41 Test #25: subdev_spec_test .................***Failed    0.01 sec
      Start 26: time_spec_test
26/41 Test #26: time_spec_test ...................***Failed    0.01 sec
      Start 27: tasks_test
27/41 Test #27: tasks_test .......................***Failed    0.01 sec
      Start 28: vrt_test
28/41 Test #28: vrt_test .........................***Failed    0.01 sec
      Start 29: expert_test
29/41 Test #29: expert_test ......................***Failed    0.01 sec
      Start 30: fe_conn_test
30/41 Test #30: fe_conn_test .....................***Failed    0.01 sec
      Start 31: eeprom_c_test
31/41 Test #31: eeprom_c_test ....................***Failed    0.01 sec
      Start 32: error_c_test
32/41 Test #32: error_c_test .....................***Failed    0.01 sec
      Start 33: ranges_c_test
33/41 Test #33: ranges_c_test ....................***Failed    0.01 sec
      Start 34: sensors_c_test
34/41 Test #34: sensors_c_test ...................***Failed    0.01 sec
      Start 35: string_vector_c_test
35/41 Test #35: string_vector_c_test .............***Failed    0.00 sec
      Start 36: subdev_spec_c_test
36/41 Test #36: subdev_spec_c_test ...............***Failed    0.00 sec
      Start 37: nocscript_expr_test
37/41 Test #37: nocscript_expr_test ..............***Failed    0.00 sec
      Start 38: nocscript_ftable_test
38/41 Test #38: nocscript_ftable_test ............***Failed    0.00 sec
      Start 39: nocscript_parser_test
39/41 Test #39: nocscript_parser_test ............***Failed    0.00 sec
      Start 40: config_parser_test
40/41 Test #40: config_parser_test ...............***Failed    0.00 sec
      Start 41: paths_test
41/41 Test #41: paths_test .......................***Failed    0.00 sec

0% tests passed, 41 tests failed out of 41
```

**Comment by amore-amore** [comment]
Any improvements to have this issue fixed?

**Comment by amore-amore** [comment]
Fixed in uhd-3.14.0.0-rc1

**Comment by michaelld** [comment]
OK, since this issue is fixed I'm closing it.
